### PR TITLE
Update ecmwf-atlas, fckit, gsibec packages for skylab v4

### LINKED ISSUE #470: Update atlas to 0.32.1 after unified-dev environment was created
**Which software in the stack would you like installed?**
New release of ecmwf-atlas 0.32.1 - see https://github.com/ecmwf/atlas/releases/tag/0.32.1 for the release

**What is the version/tag of the software?**
See above

**What compilation options would you like set?**
Same as for previous installs, no changes

**Which machines would you like to have the software installed?**
See currently managed HPC machines.

**Any other relevant information that we should know to correctly install the software??**
n/a

**Additional context**
n/a

### LINKED ISSUE #473: Update gsibec to 1.0.7 for the next release
For the next spack-stack release, we'll need a higher version of gsibec in the stack (for the use in JEDI components: saber and fv3-jedi)
Tag: probably 1.0.7, @benjaminmenetrier -- could you please confirm?

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

Update ecmwf-atlas, fckit, gsibec versions in various config files in preparation for spack-stack-1.3.0/Skylab v4 releases.

## Dependencies

- Waiting on https://github.com/NOAA-EMC/spack/pull/232

## Issues

- Fixes https://github.com/NOAA-EMC/spack-stack/issues/470
- Fixes https://github.com/NOAA-EMC/spack-stack/issues/473

### CODE REVIEW

**Comment by climbfuji** [comment]
@srherbener This PR contains the spack submodule pointer update for your CRTM hash bug fix.

File: configs/templates/skylab-dev/spack.yaml [hunk 1]
```diff
@@ -46,7 +45,7 @@ spack:
     #- gdal@3.4.3
     #- geos@3.9.1
     - gftl-shared@1.5.0
-    - gsibec@1.0.6
+    - gsibec@1.0.7
     - hdf5@1.12.1
```
Comment (srherbener) [inline configs/templates/skylab-dev/spack.yaml]: In noticed that  hdf5, netcdf-c, netcdf-fortran, parallelio in this config and skylab-no-python-dev are on older versions that what is specified in common/packages.yaml. Is that intended, or should these versions be updated here as well.

Comment (climbfuji) [reply configs/templates/skylab-dev/spack.yaml]: These should get updated! Let me do this today

**Comment by srherbener** [comment]
I tested on my arm64 Mac (apple-clang 14.0.0, gnu fortran 12.2.0, ventura os 13.1) and with my changes in #489 the spack-stack build completes successfully, as well as a subsequent build of jedi-bundle. I see some test failures but I suspect these are not related to this PR.

**Comment by climbfuji** [comment]
@srherbener I updated the versions in the skylab-dev and skylab-no-python-dev packages in commit https://github.com/NOAA-EMC/spack-stack/pull/485/commits/ff2de08d22a124cf032da61504d02976b7650968.

**Review by srherbener — APPROVED** [review]
The updates in the skylab templates look good. Thanks!

**Pull request merged**